### PR TITLE
Integration of Azure OpenAI models as third party

### PR DESCRIPTION
**Pull request by JeromeGuyon**
*Issue #380 : Azure OpenAI integration*

*Description of changes:*
Integrate Azure OpenAI model as ThirdParty

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/shared/layers/python-sdk/python/genai_core/models.py [hunk 1]
```diff
@@ -48,6 +52,22 @@ def list_openai_models():
         if model["id"].startswith("gpt")
     ]
 
+def list_azure_openai_models():
+    # azure openai model are listed, comma separated in AZURE_OPENAI_MODELS variable in external API secret
+    models = genai_core.parameters.get_external_api_key("AZURE_OPENAI_MODELS")
+
+    return [
+        {
+            "provider": Provider.AZURE_OPENAI.value,
```
Comment (bigadsoleiman) [inline lib/shared/layers/python-sdk/python/genai_core/models.py:61]: It seems this PR is missing the update of `Provider` enum to include `AZURE_OPENAI`

Comment (bigadsoleiman) [inline lib/shared/layers/python-sdk/python/genai_core/models.py:61]: It seems this PR is missing the update of `Provider` enum to include `AZURE_OPENAI`

**Review by bigadsoleiman — CHANGES_REQUESTED** [review]

File: lib/shared/layers/python-sdk/python/genai_core/models.py (hunk 1 above)
Comment (JeromeGuyon) [reply lib/shared/layers/python-sdk/python/genai_core/models.py:61]: Yes, sorry for the mistake.
I have added the change stashed the whole changes and rebased the whole branche onto main

**Review by bigadsoleiman — APPROVED** [review]

File: docs/documentation/model-requirements.md [hunk 2]
```diff
@@ -46,3 +46,31 @@ For example, if you wish to be able to interact with AI21 Labs., OpenAI's and Co
...
+- Open the SharedApiKeysSecretxyz in SecretManager
+- Update the Secrets by adding following keys to the JSON (or Key/values): 
+
+```
+{
+  "AZURE_OPENAI_MODELS": "Model1,Model2",
+  "AZURE_OPENAI_API_TYPE__Model1": "azure",
+  "AZURE_OPENAI_API_VERSION__Model1": "2023-05-15",
+  "AZURE_OPENAI_API_BASE__Model1": "https://model1.openai.azure.com/",
+  "AZURE_OPENAI_API_KEY__Model1": "7190a163395b42efb2bcb3d526ebf9a9",
```
Comment (bigadsoleiman) [inline docs/documentation/model-requirements.md:61]: can you you replace the key with a placeholder?

File: docs/documentation/model-requirements.md [hunk 3]
```diff
@@ -46,3 +46,31 @@ For example, if you wish to be able to interact with AI21 Labs., OpenAI's and Co
...
+  "AZURE_OPENAI_API_DEPLOYMENT_NAME__Model1": "Model1DeploymentName",
+  "AZURE_OPENAI_API_TYPE__Model2": "azure",
+  "AZURE_OPENAI_API_VERSION__Model2": "2023-07-01-preview",
+  "AZURE_OPENAI_API_BASE__Model2": "https://model2.openai.azure.com/",
+  "AZURE_OPENAI_API_KEY__Model2": "ffee5ba7a00b42ee851722ea44cc1826",
```
Comment (bigadsoleiman) [inline docs/documentation/model-requirements.md:66]: same here

**Review by bigadsoleiman — CHANGES_REQUESTED** [review]
Thanks for the PR! 

Added one feedback and good to go

**Review by bigadsoleiman — APPROVED** [review]

**Pull request merged**